src/Pusher.php

```diff
@@ -670,8 +674,12 @@ public function socket_auth($channel, $socket_id, $custom_data = null)
         if ($custom_data) {
             $signature['channel_data'] = $custom_data;
         }
+        if (PusherCrypto::is_encrypted_channel($channel)) {
+            $crypto = new PusherCrypto($this->settings['encryption_key']);
+            $signature['shared_secret'] = base64_encode($crypto->generate_shared_secret($channel));
+        }
 
-        return json_encode($signature);
+        return json_encode($signature, JSON_UNESCAPED_SLASHES);
```

Comment:
Does JS not unescape the slashes by default?

Reply:
json_encode escapes the slashes by defsult, which causes the JS lib to error since the string it received wasn't of the correct length which is why I did this

Reply:
Hmm I'm not sure I understand..

It appears JS does handle the escapes slashes ok:

```
>  JSON.parse("{\"test\":\"a\\/a\"}").test;
<- "a/a"
```

Reply:
You'd be better off chatting to James about why this was an issue - this sounds like it might be a fix for an issue in the PusherJS implementation.

Reply:
@leesio?

Reply:
I didn't look into it too much. I just know it worked with this and didn't without.

Reply:
Plus go by default doesn't escape slashes like the PHP one does by default, so I guess we should decide on one approach or the other

Reply:
Hmm that's strange that JS didn't seem to work when it appears to in the console...

It looks like ruby also does not escape slashes by default, e.g.

```ruby
> puts JSON.parse('{"test": "a/a"}')['test']
=> a/a
```

If Go doesn't escape too, it does appear more "standard" so I'm happy if you leave this as-is.